### PR TITLE
Improve fuzzing harness to avoid memory exhaustion

### PR DESCRIPTION
**Pull request by lopopolo**
This patch improves the fuzzing harness by explicitly limiting memory usage and maximum input length to prevent out-of-memory (OOM) errors during fuzz testing. It sets an RSS limit of 2GB and restricts maximum input string length to 2048 bytes, ensuring that buffer allocations remain within safe limits defined by the formatter's internal size limiter.

Changes include:
- Setting `-rss_limit_mb=2048` and `-max_len=2048` for the fuzz targets in the GitHub Actions workflow.
- Adding explanatory comments to clarify these limits.
- Referencing MRI Ruby's `strftime.c` for the size limit calculation rationale.
- Enhancing fuzz target implementations to include calls to formatter methods to test input handling.
- Adding explicit test cases for invalid format strings containing embedded null bytes and incomplete format specifiers.

These improvements ensure that fuzzing effectively explores edge cases without triggering resource exhaustion, aligning the testing approach more closely with MRI Ruby's handling of large or invalid inputs.

Ref: https://github.com/ruby/ruby/blob/v3_4_2/strftime.c#L921-L928

See also https://github.com/artichoke/artichoke/issues/2840 which was opened as part of this investigation.

---

The input generated by the fuzzer which caused the OOM is 3238 bytes long and expands to ~1.5GB (1,471,087,884 bytes) before reaching the invalid specifier at the last byte. The current size limiter permits a max size of 1,697,644,544 bytes so this expansion is _just_ permissible.

<details>

<summary> Fuzzer failure </summary>

```
Failing input:

	fuzz/artifacts/string/oom-d13e14abcf7a02540cd7ae6bfaaef14bd754b1b4

Output of `std::fmt::Debug`:

	(
	    MockTime {
	        year: 0,
	        month: 9,
	        day: 9,
	        hour: 9,
	        minute: 0,
	        second: 0,
	        nanoseconds: 2559,
	        day_of_week: 0,
	        day_of_year: 0,
	        to_int: 3761688987579720960,
	        is_utc: false,
	        utc_offset: 875836468,
	        time_zone: "44",
	    },
	    "44444%%%%\0#%F%%F44444444544444%F%%\0$%%0F%44544444444444444%2%%%\0#%F2%\0%%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%4\0\0\u{1}%F%%\0%%0F%4\n\n\0\t\t\t\t4\0\0\0\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%\0%%0F%44444445444444444%%%%\0#%F%%F%4440F%44544444444444444%2%%%\0#%F2%\0%%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%4\0\0\u{1}%F%%\0%%0F%4\n\n\0\t\t\t\t4\0\0\0\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%\0%%0F%44444445444444444%%%%\0#%F%%F%44444444444544444%F%%\0%%0F%4444544444%F%%\0%%0F%44444444444444444%%%%\0#%F%%%\0%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%44444445444\0\0\0\0\0\0\0\0\0\0\0\0%%\0#%F%%F%4\n\n\0\t\t\t\t\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%%F%%\0-%0F%44444445444444444%%%%\0#%F-%F%44444444444544444%F%%\0%%0F%45\0\0\0\0%\0\u{1}F%%\0%%0F%44444444444544444%%0F%4444444544444\0#%F%%%\0%%44444444444%2%%%\0#%F2%\0%%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%4\0\0\u{1}%F%%\0%%0F%4\n\n\0\t\t\t\t4\0\0\0\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%\0%%0F%44444445444444444%%%%\0#%F%%F%44444444444544444%F%%\0%%0F%4444544444%F%%\0%%0F%44444444444444444%%%%\0#%F%%%\0%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%44444445444\0\0\0\0\0\0\0\0\0\0\0\0%%\0#%F%%F%4\n\n\0\t\t\t\t\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%%F%%\0-%0F%44444445444444444%%%%\0#%F-%F%44444444444544444\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%\0\0\04444%444444t5444\0\0\0\0\0\0\0\0\0\044444%%%%44%%%%\0#%F%%%\0%%F%\0\u{2}\0%%5\0\0\0\0%\0\u{1}F%%\0%%0F%44444444444544444%%0F%4444444544444\0#%F%%%\0%%444444444%%%F%\0%\0#%F%%F%444444444445%%\0%%0F%444\u{2}44445444444444%%%%\0w%F%%F%44444444444544445%F%%\0%%44444444444544444%F%%\0%%0F%44444445444\0\0\0\0\0\0\0\0\0\0\0\0%%\0#%F%%F%4\n\n\0\t\t\t\t\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%%F%%\0-%0F%44444445444444444%%%%\0#%F-%F%44444444444544444%F%%\0%%0F%4444544444%F%%\0%%0F%44444444444444444%%%%\0#%F%%%\0%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%4444444444%444444t5444\0\0\0\0\0\0\0\0\0\044444%%%%44%%%%\0#%F%%%\0%%F%\0\u{2}\0%%5\0\0\0\0%\0\u{1}F%%\0%%0F%00000001024857295%%0F%4444444544444\0#%F%%%\0%%444444444%%%F44%%%F%\0%\0#%F%%F%444444444445%%\0%%0F%444\u{2}44445444444444%%%%\0w%F%%F%44444444444544445%F%%\0%%4444444444444%%%%\0\0-%0F%44444445444444444%%%%\0#%F-%F%44444444444544444%F%%\0%%0F%4444544444%F%%\0%%0F%44444444444444444%%%%\0#%F%%%\0%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%4444444444%444444t5444\0\0\0\0\0\0\0\0\0\044444%%%44444444544444%F%%\0%%0F%4444544444%F%%\0%%0F%44444444444444444%%%%\0#%F%%%\0%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%44444445444\0\0\0\0\0\0\0\0\0\0\0\0%%\0#%F%%F%4\n\n\0\t\t\t\t\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%%F%%\0-%0F%44444445444444444%%%%\0#%F-%F%44444444444444444%444444t5444\0\0\0\0\0\0\0\0\0\044444%%%44444444544444%F%%\0%%0F%4444544444%F%%\0%%0F%44444444444444444%%%%\0#%F%%%\0%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%44444445444\0\0\0\0\0\0\0\0\0\0\0\0%%\0#%F%%F%4\n\n\0\t\t\t\t\0\0\0\0\0\0\0\0\0\0\0~\0%%%\0\u{1}%F%%\0%%0F%44444444444544444%F%%\0%%0F%444444444444544444%F%%%F%%\0-%0F%44444445444444444%%%%\0#%F-%F%44444444444544444%F%%\0%%0F%4444544444%F%%\0%%0F%44444444444444444%%%%\0#%F%%%\0%%F%\0\u{2}\0%%0\0\0\0\0\0\u{1}%F%%\0%%0FF%%\0%%0F%44444444444544444%%0F%4444444544444\0#%F%%%\0%%444444444%%%F%\0%\0#%F%%F%444444444445%%\0%%0F%444\u{2}44445444444444%%%%\0w%F%%F%44444444444544445%F%%\0%%4444444444444%%%%\0\0-%0F%44444445444444444%%%%\0#%F-%F%444444444%%%%\0#%F%%%\0%%F%",
	)
```

</details>

### CODE REVIEW

File: fuzz/fuzz_targets/bytes.rs [hunk 1]
```diff
@@ -7,6 +7,8 @@ use mock::MockTime;
 
 fuzz_target!(|data: (MockTime, &[u8])| {
     let (time, format) = data;
+    let _ignored = strftime::bytes::strftime(&time, format, &mut buf[..]);
```
Comment (x-hgg-x) [inline fuzz/fuzz_targets/bytes.rs]:
```suggestion
    let _ignored = strftime::bytes::strftime(&time, format);
```

File: fuzz/fuzz_targets/string.rs [hunk 2]
```diff
@@ -30,6 +30,8 @@ impl<'a> fmt::Write for LimitedBuf<'a> {
 
 fuzz_target!(|data: (MockTime, &str)| {
     let (time, format) = data;
+    let _ignored = strftime::string::strftime(&time, format, &mut buf[..]);
```
Comment (x-hgg-x) [inline fuzz/fuzz_targets/string.rs]:
```suggestion
    let _ignored = strftime::string::strftime(&time, format);
```

**Review by x-hgg-x — APPROVED** [review]
